be/src/exec/vectorized/analytic_node.cpp

```diff
@@ -175,7 +175,8 @@ Status AnalyticNode::_get_next_for_unbounded_preceding_range_frame(RuntimeState*
         while (_analytor->current_row_position() < _analytor->partition_end() &&
                _analytor->window_result_position() < chunk_size) {
             if (_analytor->current_row_position() >= _analytor->peer_group_end()) {
-                _analytor->find_peer_group_end();
+                DCHECK_EQ(_analytor->partition_end(), _analytor->found_partition_end());
```

Comment:
why `_analytor->partition_end()` equal to `analytor->found_partition_end()` ?

Reply:
In non-pipeline, range window frame won't be evaluated unless the whole partition is reached. if the condition (_analytor->current_row_position() < _analytor->partition_end()) of while loop is satisfied, which means `_partition_start <= _current_row_position < _partition_end`, so `_partition_end` must pointer at the genuine partition boundary, and `_found_partition_end` won't be greater than `_partition_end` if current partition haven't finished evaluating

```
bool Analytor::find_and_check_partition_end() {
    // current partition data don't consume finished
    if (_current_row_position < _partition_end) {
        _found_partition_end = _partition_end;
        return false;
    }

    // ...
}
```

Reply:
I add some comments